.claude/hooks/Stop/invoke_session_validator.py

```
@@ -128,6 +128,56 @@ def get_missing_keys(log_content: str) -> list[str]:
+    try:
+        skip_log = os.path.join(project_dir, ".agents", "sessions", "session-end-skips.jsonl")
+        os.makedirs(os.path.dirname(skip_log), exist_ok=True)
+        record = json.dumps({
+            "timestamp": datetime.now(tz=UTC).isoformat(),
+            "event": "session_closed_without_session_end",
+            "session_id": session_id,
+            "session_log": session_log_name,
+        })
+        with open(skip_log, "a", encoding="utf-8") as f:
+            f.write(record + "\n")
+    except Exception as exc:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)